content/en/blog/_posts/2022-04-12-prevent-unauthorised-volume-mode-conversion.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: blog
+title: 'Prevent unauthorised volume mode conversion'
+date: 2022-04-12
+slug: prevent-unauthorised-volume-mode-conversion-alpha
+---
+
+**Author:** Raunak Shah (Mirantis)
+
+Kubernetes v1.24 introduces a new alpha-level feature that prevents unauthorised users from modifying the volume mode of a [`PeristentVolumeClaim`](/docs/concepts/storage/persistent-volumes.md). This feature requires [`VolumeSnapshot`](/docs/concepts/storage/volume-snapshots.md) APIs with version `v6.0.0` onwards and `external-provisioner` version `v3.2.0` onwards.  
+
+### The problem
+
+As of Kubernetes 1.23, users can leverage the `VolumeSnapshot` feature, which GA'd in Kubernetes 1.20, to create a `PersistentVolumeClaim` (or `PVC`) from a previously taken `VolumeSnapshot`. This is done by pointing the `Spec.dataSource` parameter of the `PVC` to an existing `VolumeSnapshot` instance. 
+There is no logic that validates whether the original volume mode of the `PVC`, whose snapshot was taken, matches the volume mode of the newly created `PVC`, that is being created from the existing `VolumeSnapshot`.
+
+There is logic in allowing this, as many popular storage backup vendors convert the volume mode, during the course of a backup operation, for efficiency purposes.
+
+However this also presents a security gap that allows malicious users to potentially exploit an as-yet-unknown CVE in the kernel.
```

Comment:
```suggestion
However this also presents a security gap that allows malicious users to potentially exploit an as-yet-unknown flaw in a node's operating system.
```

Reply:
Changed to `vulnerability in the host operating system.`